policy.go

```diff
@@ -42,6 +42,155 @@ var (
 	lockNVIndexAttrs = tpm2.NVTypeOrdinary.WithAttrs(tpm2.AttrNVPolicyWrite | tpm2.AttrNVAuthRead | tpm2.AttrNVNoDA | tpm2.AttrNVReadStClear)
 )
 
+// policyPCRParam details a set of approved digests for a given PCR.
+type policyPCRParam struct {
+	pcr     int
+	alg     tpm2.HashAlgorithmId
+	digests tpm2.DigestList
+}
+
+// dynamicPolicyComputeParams provides the parameters to computeDynamicPolicy.
+type dynamicPolicyComputeParams struct {
+	key *rsa.PrivateKey // Key used to authorize the generated dynamic authorization policy
+
+	// signAlg is the digest algorithm for the signature used to authorize the generated dynamic authorization policy. It must
+	// match the name algorithm of the public part of key that will be loaded in to the TPM for verification.
+	signAlg              tpm2.HashAlgorithmId
+	pcrParams            []policyPCRParam // Approved PCR digests
+	policyCountIndexName tpm2.Name        // Name of the NV index used for revoking authorization policies
+
+	// policyCount is the maximum permitted value of the NV index associated with policyCountIndexName, beyond which, this authorization
+	// policy will not be satisfied.
+	policyCount uint64
+}
+
+type policyPCRData struct {
+	PCR       int
+	Alg       tpm2.HashAlgorithmId
+	OrDigests tpm2.DigestList
+}
+
+type dynamicPolicyData struct {
+	PCRData                   []policyPCRData
+	PolicyCount               uint64
+	AuthorizedPolicy          tpm2.Digest
+	AuthorizedPolicySignature *tpm2.Signature
+}
+
+// staticPolicyComputeParams provides the parameters to computeStaticPolicy.
+type staticPolicyComputeParams struct {
+	key                  *rsa.PublicKey  // Public part of key used to authorize a dynamic authorization policy
+	pinIndexPub          *tpm2.NVPublic  // Public area of the NV index used for the PIN
+	pinIndexAuthPolicies tpm2.DigestList // Metadata for executing policy sessions to interact with the PIN NV index
+	lockIndexName        tpm2.Name       // Name of the global NV index for locking access to sealed key objects
+}
+
+// staticPolicyData is an output of computeStaticPolicy and provides metadata for executing a policy session.
+type staticPolicyData struct {
+	AuthorizeKeyPublic   *tpm2.Public
```

Comment:
AuthorizeKey read strangely? is it an AuthorisationKey or something else?

Reply:
Authorization/authorisation might be more correct, but AuthorisationKeyPublic seems a bit verbose - would AuthKeyPublic be better?

Reply:
yes,  the code seems to use sometimes  "...PublicKey"  and "KeyPublic" what's difference?  should it use only one of this two?   (snapd itself tend to use PrivateKey/PublicKey like go own libs)